### PR TITLE
Follow user draft, users' scrolls follow the other's cursor around the same file. Works for clicking and arrow-key movement.

### PR DESCRIPTION
**Pull request by DaCatDude**
Please let me know if this works, I did some testing in a long javascript file, but I havent tested it anywhere else. As far as I know, it should work without many issues.

### CODE REVIEW

**Comment by DaCatDude** [comment]
The next goal here would be to add some sort of toggle, so that further work doesnt get confusing

File: src/client/CodeEditor/json1PresenceDisplay.ts [hunk 1]
```diff
@@ -36,8 +40,14 @@ export const json1PresenceDisplay = ({
...
         const presenceState = {};
 
+        // Add the scroll event listener
+        //This runs for the arrow key scrolling, it should result in the users scrolling to eachother's location.
+        view.dom.addEventListener('scroll', () => {
+          this.scrollToCursor(view);
+        });
         // Receive remote presence changes.
         docPresence.on('receive', (id, presence) => {
+          console.log(`Received presence for id ${id}`, presence); // Debug statement
```
Comment (curran) [inline src/client/CodeEditor/json1PresenceDisplay.ts:50]:
```suggestion
          // console.log(`Received presence for id ${id}`, presence); // Debug statement
```

File: src/client/CodeEditor/json1PresenceDisplay.ts [hunk 2]
```diff
@@ -97,6 +107,12 @@ export const json1PresenceDisplay = ({
                 }),
               });
             }
+            if (view.state.doc.length >= from) { // Ensure position is valid
+              this.cursorPosition[id] = from; // Store the cursor position, important to run if we cant get the regular scroll to work
+              console.log(`Stored cursor position for id ${id}: ${from}`); // Debug statement
```
Comment (curran) [inline src/client/CodeEditor/json1PresenceDisplay.ts:112]:
```suggestion
              // console.log(`Stored cursor position for id ${id}: ${from}`); // Debug statement
```

File: src/client/CodeEditor/json1PresenceDisplay.ts [hunk 3]
```diff
@@ -97,6 +107,12 @@ export const json1PresenceDisplay = ({
...
+            } else {
+              console.warn(`Invalid cursor position for id ${id}: ${from}`); // Debug statement
```
Comment (curran) [inline src/client/CodeEditor/json1PresenceDisplay.ts:114]:
```suggestion
              // console.warn(`Invalid cursor position for id ${id}: ${from}`); // Debug statement
```

**Review by curran — APPROVED** [review]
LGTM!

**Pull request merged**